### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by leggomuhgreggo** [leggomuhgreggo/code-good#2]
Re: Comments

They’re just so easily abused…

I agree they are virtuous in their capacity to enrich understanding, but:

1. Comments are _fundamentally_ a secondary, non-point-source representation of meaning. Because they are derivative of the code, any code changes compel an additional phase to consider the corresponding comments, which is often times a pro forma exercise that is potentially extra tedious because not everyone has good english language skills.
2. They’re a bandaid for poorly designed code, symptomatic of SRP-violations, and interfaces ambiguity, which is better alleviated by semantically meaningful method/parameter naming, clear control flow signals.

It’s true that sometimes you can’t adequately represent all the material context through rich naming and code organization etc.

## Comments

TL;DR: Comments are often a sign of imperative code

muddled, disordered, rudiment, nuts and bolts, byzantine, enigmatic, bewildering,  

we don't code in zeros and ones

Heading vs paragraphs

## DSLs

Every API a developer has to learn costs. Don’t do unless you’re writing a formal tool, with a mature enough API to justify.  Don’t be indulgent. 

### arcane

adjective - known or understood by very few; mysterious; secret; obscure; esoteric:

> She knew a lot about Sanskrit grammar and other arcane matters.
